### PR TITLE
Move team stats into team manager

### PR DESCRIPTION
**Pull request by c-bartell**
## Moved TeamStatistics module functionality into TeamManager
 - Created tests for all methods
 - Created and tested relevant helper methods
 - Pulled down recent merges and resolved conflicts / debugged

**All tests passing, spec harness passing.**

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -3,19 +3,19 @@
 require_relative 'team_manager'
 
 class StatTracker
-  attr_reader :game_team_manager, :game_manager, :team_manager
-  def self.from_csv(locations)
+  attr_reader :game_teams_manager, :game_manager, :team_manager
+  def self.from_csv(locations) # I need a test
```
Comment (s-steel) [inline lib/stat_tracker.rb:7]: Definitely good to look into some tests.

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -51,8 +51,50 @@ def average_goals_by_season
     @game_manager.average_goals_by_season
   end
 
+  # Team Stats
```
Comment (s-steel) [inline lib/stat_tracker.rb:54]: I like having each section labeled.

**Review by cpfergus1 — COMMENTED** [review]
Looks great! Great re-org

File: test/game_team_manager_test.rb [hunk 3]
```diff
@@ -83,41 +83,89 @@ def test_team_accuracy
...
+    game_team_array = [game_team1, game_team2, game_team3]
+    CSV.stubs(:foreach).returns(nil)
+    game_team_manager = GameTeamManager.new('A totally legit path', stat_tracker)
+    game_team_manager.stubs(:game_teams).returns(game_team_array)
+
+    assert_equal ['1', '2'], game_team_manager.game_ids_by_team('1')
+    assert_equal ['3'], game_team_manager.game_ids_by_team('2')
+  end
+
+  def test_it_can_fetch_game_team_info
```
Comment (s-steel) [inline test/game_team_manager_test.rb:144]: Nice work with mocks and stubs.

**Review by s-steel — CHANGES_REQUESTED** [review]
Overall looks good.  Just one thing.  On the stat_tracker file we are using game_team_manager, not game_teams_manager.  Sorry for the confusion! It should be fine everywhere else, just in the stat_tracker.

**Comment by c-bartell** [comment]
@s-steel  Good call out! I had to pick one when I was debugging to get the tests to work, and it looks like I picked wrong lol.  I'll do a quick edit and submit in another PR.

**Pull request merged**